dspace-spring-rest/src/test/java/org/dspace/app/opensearch/OpenSearchControllerTest.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.opensearch;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.xpath;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+
+import org.junit.Test;
+
+/**
+ * Integration test to test the /opensearch endpoint
+ * (Class has to start or end with IT to be picked up by the failsafe plugin)
+ *
+ * @author Oliver Goldschmidt (o dot goldschmidt at tuhh dot de)
+ */
+public class OpenSearchControllerTest extends AbstractControllerIntegrationTest {
+
+    @Test
+    public void searchAtomTest() throws Exception {
+        //When we call the root endpoint
+        getClient().perform(get("/opensearch/search")
+                                .param("query", "cats"))
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/atom+xml;charset=UTF-8"
+                   .andExpect(content().contentType("application/atom+xml;charset=UTF-8"))
```

Comment:
can you extend the test a bit? we should at least test the basic use case creating some items that will match our search. You should check that these items are included in the response.
It will be useful to create separate test methods for edge situation like:
- query without results
- invalid query: is this defined in the specs?
- rich SOLR query: to be sure that these are supported over openSearch or otherwise document that special solr characters are automatically escaped

Reply:
Yes, I have forgotten to mention, that I have already extended the tests. IMO this is enough, but if you think, there is still a need for more extensions, just let me know :-)